### ISSUE TITLE
ctx of cog_context_menu is of type SlashContext, instead of MenuContext

### ISSUE THREAD

**Issue by Mayerch1** [interactions-py/interactions.py#291]
When using context menus within a cog, the type of the `ctx` object is `SlashContext`, instead of `MenuContext`.
The `ctx.data` field will hold the attributes `target_id` and `resolved` as they would be required for a conversion to `MenuContext`

Reproduce with this snippet in the bots main-file
```
@slash.context_menu(name='TEST_CMD_1', guild_ids=[GUILD_ID], target=ContextMenuType.USER)
async def incident_report_context(ctx: MenuContext):
    print(f'@slash {type(ctx)}')
```

and with this snippet within a cog

```
@cog_ext.cog_context_menu(name='TEST_CMD_2', guild_ids=[GUILD_ID], target=ContextMenuType.USER)
async def incident_report_context_2(self, ctx: MenuContext):
    print(f'@cog_ext {type(ctx)}')
```

The console output shows following output on triggering
```
<class 'discord_slash.context.MenuContext'>
<class 'discord_slash.context.SlashContext'>
```

**Comment by i0bs** [comment]
This is a currently known issue

**Comment by i0bs** [comment]
Fixed in #293.
